### PR TITLE
Remove qt4agg backend references

### LINKED ISSUE #32426: Remove Qt4 Matplotlib Backend
In the codebase there are numerous places where we are using Qt4. These need to be removed as new build servers do not include Qt4 as we haven't supported this since Mantid 5.x

In each instance we should check, in order of preference:
- Can we remove it (dead code)
- Can we replace it with https://github.com/mantidproject/mantid/blob/master/qt/python/mantidqt/MPLwidgets.py
`python 
- Can we hardcode it to Qt5Agg

### PR DESCRIPTION
**Pull request by martyngigg**
**Description of work.**

Removes all `qt4agg` and related references from the code as we no longer support Qt4

**To test:**

* Check the `PyChop` GUI under `Interfaces->Direct` still starts

Fixes #32426 

*This does not require release notes* because **it is an internal change.**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by SilkeSchomann — DISMISSED** [review]
I can confirm that the PyChop GUI still works after these changes.

**Review by DavidFair — COMMENTED** [review]
LGTM, just one place where I think we've missed some Qt4 stuff?

File: qt/python/mantidqt/mantidqt/gui_helper.py [hunk 1]
```diff
@@ -27,16 +27,12 @@ def set_matplotlib_backend():
     correct matplotlib widgets.'''
     backend = matplotlib.get_backend()
     if backend.startswith('module://'):
-        if backend.endswith('qt4agg'):
-            backend = 'Qt4Agg'
-        elif backend.endswith('workbench') or backend.endswith('qt5agg'):
+        if backend.endswith('workbench') or backend.endswith('qt5agg'):
             backend = 'Qt5Agg'
     else:
         from qtpy import PYQT4, PYQT5  # noqa
```
[the remark this answers is not in the document]
Comment (martyngigg) [reply qt/python/mantidqt/mantidqt/gui_helper.py:33]: Ah yes. I missed the light gray text in VS Code and the `noqa` prevented flake8 from warning me 😞  I'll get rid of this too.

**Comment by martyngigg** [comment]
That should be gone now.

**Review by SilkeSchomann — APPROVED** [review]
PyChop still works as expected.

**Review by DavidFair — APPROVED** [review]
Thanks :smile:

**Pull request merged**